### PR TITLE
[skip ci] Add devEngines to package.json

### PR DESCRIPTION
**Pull request by yavorsky**
| Q                        | A 
| ------------------------ | ---
| Patch: Bug Fix?          |  n
| Major: Breaking Change?  |  n
| Minor: New Feature?      |  n
| Deprecations?            |  n
| Spec Compliancy?         |  n
| Tests Added/Pass?        |  n
| Fixed Tickets            | 
| License                  | MIT
| Doc PR                   | n
| Dependency Changes       | n

`devEngines` is a good way to show environment where build script is expected to work.
Also, going to add for 7.0 brunch without 0.12.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,6 +46,10 @@
     "through2": "^2.0.0",
     "uglify-js": "^2.4.16"
   },
+  "devEngines": {
+    "node": "0.12 || 4.x || 5.x || 6.x || 7.x",
```
Comment (xtuc) [inline package.json]: We have drop node 0.12. I guess we can remove it.

Comment (yavorsky) [inline package.json]: Current version from master builds on 0.12 🤔 . All works except linting

Comment (hzoo) [inline package.json]: We support 0.10 on 6.x as well (circle ci) https://github.com/babel/babel/blob/master/circle.yml which we removed in 7

Comment (danez) [inline package.json]: Though I wouldn't suggest to anyone to develop babel on node <4. I would say for developing people should use latest or LTS.

Comment (hzoo) [inline package.json]: For our own development? Oh yeah definetely at least 6

Comment (yavorsky) [inline package.json]: devEngines is not about recommendations but just shows where stuff could technically work

Comment (yavorsky) [inline package.json]: Maybe once preset-env will support engines and we drop es2015 in favor of env. 😏

**Comment by xtuc** [comment]
React uses the `devEngines` key for checking the environment https://github.com/facebook/react/blob/master/package.json#L91

Should we also use this?

**Comment by yavorsky** [comment]
@xtuc yeah, I thought it as a separate PR. I don't know what's better, script from fbjs or make new for own needs.

**Comment by xtuc** [comment]
@yavorsky I think the FB script is fine ([see source](https://github.com/facebook/fbjs/blob/master/packages/fbjs-scripts/node/check-dev-engines.js#L29-L62)).

This will make a new devDependency, but we could use other work they done.

**Pull request merged**